metadata-ws/src/test/java/uk/ac/ebi/ampt2d/metadata/cucumber/CommonSteps.java

```diff
@@ -0,0 +1,349 @@
+        LocalDate days = LocalDate.now().plusDays(diff);
+        CommonStates.getResultActions().andExpect(jsonPath("$."+field).value(days.toString()));
+    }
+
+    @Then("^the result should have (.*) existing$")
+    public void checkResponseJsonFieldValueExist(String field) throws Exception {
+        CommonStates.getResultActions().andExpect(jsonPath("$."+field).exists());
+    }
+
+    @Then("^the result should have (.*) non empty$")
+    public void checkResponseJsonFieldValueNotEmpty(String field) throws Exception {
+        CommonStates.getResultActions().andExpect(jsonPath("$."+field).isNotEmpty());
+    }
+
+    @Then("^the result should not contain (.*)$")
+    public void checkResponseJsonNoField(String field) throws Exception {
+        CommonStates.getResultActions().andExpect(jsonPath("$."+field).doesNotExist());
+    }
+
+    @And("^the href of the class (.*) should be (.*)$")
+    public void checkResponseLinkedObjectHref(String className, String valueKey)
+            throws Exception {
+        CommonStates.getResultActions().andExpect(jsonPath("$.."+className+".href")
+                .value(CommonStates.getUrl(valueKey)));
+    }
+
+    @And("^the href of the (.*) of (.*) has items (.*)$")
+    public void checkResponseLinkedObjectHref(String field, String className, String urlKeys)
+            throws Exception {
+        String[] urls = {};
+        if (!urlKeys.equals("NONE")) {
+            urls = Arrays.stream(urlKeys.split(","))
+                    .map(key -> CommonStates.getUrl(key))
+                    .toArray(String[]::new);
+        }
+
+        CommonStates.getResultActions().andExpect(jsonPath("$.."+className+"[*].."+field+".href",
+                containsInAnyOrder(urls)));
+    }
+
+    @And("^the (.*) field of (.*) (\\d*) should be (.*)$")
+    public void checkResponseLinkedObjectFieldValue(String field, String className, int index, String fieldValue)
+            throws Exception {
+        CommonStates.getResultActions()
+                .andExpect(jsonPath("$.."+className+"["+index+"]."+field).value(fieldValue));
+    }
+
+    @And("^the (.*) field of (.*) (\\d*) should have item (.*)$")
+    public void checkResponseLinkedObjectFieldItem(String field, String className, int index, String fieldValue)
+            throws Exception {
+        CommonStates.getResultActions()
+                .andExpect(jsonPath("$.."+className+"["+index+"]."+field).isArray())
+                .andExpect(jsonPath("$.."+className+"["+index+"]."+field+"[*]", hasItems(fieldValue)));
+    }
+
+    @When("^user request OPTIONS with header (.*) value (.*)$")
+    public void performOptionsWithData(String headerList, String valueList) throws Exception {
+        String[] header = headerList.split(",");
+        String[] value = valueList.split(",");
+
+        CommonStates.setResultActions(mockMvc.perform(options("/")
+                .header(header[0], value[0])
+                .header(header[1], value[1])));
```

Comment:
This is not safe. You can't be sure in the future this method is used differently. In fact currently I found this method is used only in one place. Do you really have to parameterise the method? Can you design the step definition differently?